packages/web3torrent/src/clients/payment-channel-client.ts

```diff
@@ -41,7 +42,7 @@ if (process.env.REACT_APP_FAKE_CHANNEL_PROVIDER === 'true') {
 // The beneficiary proposes the channel, but accepts payments
 // The payer joins the channel, and makes payments
 export class PaymentChannelClient {
-  channelCache: Record<string, ChannelState> = {};
+  channelCache: Record<string, ChannelState | undefined> = {};
```

Comment:
Why do we need this? Is there a situation where you want to store the `channelId` but do not yet have a state for it?

Reply:
This is for type safety -- if you access `this.channelCache['someRandomString']`, you get a value of type `ChannelState` even though the value is probably `undefined`.

Reply:
Good point. https://fnune.com/typescript/2019/01/30/typescript-series-1-record-is-usually-not-the-best-choice/

Reply:
There are some other places we could think about fixing this, including most places where we do `Record<string,T>` and https://github.com/statechannels/monorepo/blob/20112d7a213340b42429247fc195b885873fc599/packages/web3torrent/src/components/torrent-info/channels-list/ChannelsList.tsx#L14